cfn-resources/private-endpoint/cmd/resource/steps/privateendpoint/private_endpoint.go

```diff
@@ -82,43 +158,70 @@ func ValidateCreationCompletion(mongodbClient *mongodbatlas.Client, groupID stri
 		return nil, &pe
 	}
 
-	privateEndpointResponse, response, err := mongodbClient.PrivateEndpoints.GetOnePrivateEndpoint(context.Background(),
-		groupID,
-		ProviderName,
-		callBackContext.ID,
-		callBackContext.InterfaceID)
-	if err != nil {
-		pe := progress_events.GetFailedEventByResponse(fmt.Sprintf("Error validating private endpoint create : %s", err.Error()),
-			response.Response)
-		return nil, &pe
+	completed := true
+	ids := make([]string, len(callBackContext.PrivateEndpoints))
+
+	for i := range callBackContext.PrivateEndpoints {
+		ids[i] = callBackContext.PrivateEndpoints[i].InterfaceEndpointID
+		if callBackContext.PrivateEndpoints[i].Status != StatusAvailable {
+			privateEndpointResponse, response, err := mongodbClient.PrivateEndpoints.GetOnePrivateEndpoint(context.Background(),
+				groupID,
+				ProviderName,
+				callBackContext.ID,
+				callBackContext.PrivateEndpoints[i].InterfaceEndpointID)
+			if err != nil {
```

Comment:
[q] why if the status is not available you perform this call?

Reply:
there are multiple endpoints in the service, if the status is available there is no need to check the status anymore, if is pending we must continue wating